ssh-key-algo

```diff
@@ -34,7 +47,12 @@ fi
 echo "using: $(which ssh)"
 
 # Disable connection reuse because we can't parse the output if that happens.
-ssh -vvv -oControlMaster=no -oControlPath=no "$host" >"$DIR/output" 2>&1
+if [ -n "$identity" ]
+then
+  ssh -vvv -oControlMaster=no -oControlPath=no -i "$identity" "$host" >"$DIR/output" 2>&1
+else
+  ssh -vvv -oControlMaster=no -oControlPath=no "$host" >"$DIR/output" 2>&1
```

Comment:
I think we could also write this maybe like this:

```
ssh -vvv -oControlMaster=no -oControlPath=no ${identity:+-i "$identity"} "$host" >"$DIR/output" 2>&1
```

That might minimize the duplication here.

Reply:
Ooh, that's better indeed.  Updated with co-author credit ... thanks!